modules/rostests/apitests/user32/SetFocus.c

```diff
@@ -0,0 +1,209 @@
+/*
+ * PROJECT:     ReactOS API tests
+ * LICENSE:     LGPL-2.1+ (https://spdx.org/licenses/LGPL-2.1+)
+ * PURPOSE:     Test for SetFocus/GetFocus/GetGUIThreadInfo
+ * COPYRIGHT:   Copyright 2021 Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
+ */
+
+#include "precomp.h"
+#include <dlgs.h> // psh1, ...
+#include <process.h>
+
+#define INTERVAL 80
+
+static DWORD s_dwMainThreadID;
+
+static HWND GetMainThreadFocus(void)
+{
+    GUITHREADINFO gui = { sizeof(gui) };
+    GetGUIThreadInfo(s_dwMainThreadID, &gui);
+    return gui.hwndFocus;
+}
+
+static unsigned __stdcall thread_proc_0(void *arg)
+{
+    HWND hwnd = arg;
+
+    ok_int(GetFocus() == NULL, TRUE);
+    ok_int(GetMainThreadFocus() == GetDlgItem(hwnd, IDOK), TRUE);
+    PostMessageA(hwnd, WM_NEXTDLGCTL, FALSE, FALSE);
```

Comment:
Why do you use postmessage here?
If you use SendMessage, the message will be processed before the function returns.

Reply:
Thanks.